### PR TITLE
Doc/43 ecs documentation

### PR DESCRIPTION
**Pull request by CedricCORGE**

### CODE REVIEW

File: src/ECS/Components/DrawableComponent.hpp [hunk 1]
```diff
@@ -13,6 +13,10 @@
 
 #include <string>
 
+/**
+ * @brief Define the texture rect to draw
+ * 
+ */
 struct Vector4 {
```
Comment (tbellicha) [inline src/ECS/Components/DrawableComponent.hpp:20]: Why a Vector4 only contains 2 members ?

File: src/ECS/Registry.hpp [hunk 2]
```diff
@@ -23,18 +23,39 @@
...
+ * @brief Storage of all sparse_array
+ * 
+ */
 class registry {
   public:
+
+    /**
+     * @brief Register a new system into the system sparse array
+     * 
+     * @param func
```
Comment (tbellicha) [inline src/ECS/Registry.hpp:36]: What is a "func" ?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -78,7 +78,7 @@ jobs:
 
       - name: Build rtype binaries on Raspberry (desktop)
         run: |
-          mkdir -p build; cd build && conan install .. --build=missing  -s compiler.libcxx=libstdc++11 -c tools.system.package_manager:mode=install -c tools.system.package_manager:sudo=True && cmake -DCMAKE_BUILD_TYPE=Debug .. && cmake --build  . -- -j
+          mkdir -p build; cd build && conan install .. --build=missing  -s compiler.libcxx=libstdc++11 -c tools.system.package_manager:mode=install -c tools.system.package_manager:sudo=True && cmake -DCMAKE_BUILD_TYPE=Debug .. && cmake --build  . -- -j1
```
[the remark this answers is not in the document]
Comment (CedricCORGE) [reply .github/workflows/build.yml:81]: without this build fail on raspberry probably due to too many core use

[the remark this answers is not in the document]
Comment (CedricCORGE) [reply .github/workflows/build.yml:81]: so i limit to 1 core

**Review by tbellicha — CHANGES_REQUESTED** [review]

**Review by JohanCDev — APPROVED** [review]

**Review by tbellicha — APPROVED** [review]

**Pull request merged**